test/mpiexecjl.jl

```diff
@@ -15,25 +15,21 @@ using MPI
         # Test a run of mpiexec
         nprocs_str = get(ENV, "JULIA_MPI_TEST_NPROCS", "")
         nprocs = nprocs_str == "" ? clamp(Sys.CPU_THREADS, 2, 4) : parse(Int, nprocs_str)
-        mpiexecjl = joinpath(dir, "mpiexecjl")
+        env = ["JULIA_BINDIR" => Sys.BINDIR]
+        mpiexecjl = addenv(`$(joinpath(dir, "mpiexecjl"))`, env...)
```

Comment:
I'm still not sure about why `JULIA_BINDIR` is _needed_, but I think that ensuring `mpiexecjl` is consistently called with the same environment in all these tests is a good change: it's less error-prone as it avoids missing something in possible future tests.  As a side effect, we can slightly simplify the code in these tests whenever we call `mpiexecjl`.